### ISSUE TITLE
feat(destinations): Add support for `migrate_mode: forced` to all destinations

### ISSUE THREAD

**Issue by erezrokah** [cloudquery/cloudquery#7993]
TLDR: Similar to https://github.com/cloudquery/cloudquery/pull/7819, but for other destinations based one each destination capabilities.

> Also see initial issue https://github.com/cloudquery/cloudquery/issues/6600

### TODO

- [ ] BigQuery - Most important I think
- [ ] Azure Blob Storage
- [ ] Elasticsearch
- [ ] File
- [ ] Firehose
- [ ] GCS
- [ ] Kafka
- [x] Microsoft SQL Server - https://github.com/cloudquery/cloudquery/pull/8318
- [ ] MongoDB
- [ ] Neo4j
- [ ] S3
- [ ] Snowflake
- [x] SQLite - https://github.com/cloudquery/cloudquery/pull/8166
- [x] ClickHouse - https://github.com/cloudquery/cloudquery/pull/8255
- [x] PostgreSQL - https://github.com/cloudquery/cloudquery/pull/7819
- [x] MySQL - https://github.com/cloudquery/cloudquery/pull/8438

The process should be to remove `SkipMigrateOverwriteForce: true` and `SkipMigrateAppendForce: true` (or only remove `SkipMigrateAppendForce: true` if the plugin doesn't support overwrite) from the tests, and ensure they pass, for example:
https://github.com/cloudquery/cloudquery/blob/fd24a5b0afc3c260827931a326b43a6973677ff7/plugins/destination/bigquery/client/client_test.go#L24

**Comment by erezrokah** [comment]
We should also update https://www.cloudquery.io/docs/advanced-topics/migrations to reflect the new behavior

**Comment by erezrokah** [comment]
Going to close this for now. We haven't seen users requests to support the remaining destinations on this list.
We can re-visit if needed
